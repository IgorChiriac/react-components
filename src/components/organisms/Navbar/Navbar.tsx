import React from 'react';

import NavbarDropdown from './NavbarDropdown/NavbarDropdown';
import NavbarLayout from './NavbarLayout/NavbarLayout';
import NavbarLink from './NavbarLink/NavbarLink';

class Navbar extends React.Component {
  public static Layout = NavbarLayout;
  public static Dropdown = NavbarDropdown;
  public static Link = NavbarLink;

  public render() {
    return null;
  }
}

export default Navbar;
