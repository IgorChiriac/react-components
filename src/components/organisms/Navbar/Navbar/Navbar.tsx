import React from 'react';
import styled from 'styled-components';

import { colors } from '../../../../constants/colors';
import { navbarHeight } from '../../../../constants/components';
import useScrollThreshold from '../useScrollThreshold/useScrollThreshold';
import FlexContainer from '../../../layout/FlexContainer/FlexContainer';

export interface ILayoutOuterProps extends React.HTMLAttributes<HTMLDivElement> {
  backgroundColor: string;
  overlap: boolean;
}

const LayoutOuter = styled.div<ILayoutOuterProps>`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  background-color: ${({ backgroundColor }) => backgroundColor};
  height: ${navbarHeight}px;
  transition: box-shadow 0.2s ease-in-out;
  box-shadow: 0 0 0 transparent;
  ${({ overlap }) => overlap && `box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px;`}
`;

const LayoutInner = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-between;
  max-width: 960px;
  height: 100%;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
`;

const Spacer = styled.div`
  height: ${navbarHeight}px;
`;

const Container = styled(FlexContainer)`
  height: inherit;
`;

export interface INavbarProps {
  /**
   * background color
   */
  backgroundColor?: string;
  /**
   * Component appearing on the left
   */
  left?: React.ReactNode;
  /**
   * Component appearing in the center
   */
  center?: React.ReactNode;
  /**
   * Component appearing on the right
   */
  right?: React.ReactNode;
}

const Navbar = ({ backgroundColor = colors.neutral.white, left, center, right }: INavbarProps) => {
  const overThreshold = useScrollThreshold();

  return (
    <>
      <Spacer />
      <LayoutOuter backgroundColor={backgroundColor} overlap={overThreshold}>
        <Container>
          <LayoutInner>
            {left && <div>{left}</div>}
            {center && <div>{center}</div>}
            {right && <div>{right}</div>}
          </LayoutInner>
        </Container>
      </LayoutOuter>
    </>
  );
};

export default Navbar;
