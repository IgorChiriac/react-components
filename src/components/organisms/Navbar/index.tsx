import React, { FC } from 'react';

import Navbar, { NavbarProps } from './Navbar/Navbar';
import NavbarAction from './NavbarAction/NavbarAction';
import { OpenProvider } from './OpenProvider';

interface NavbarStatic {
  Action: typeof NavbarAction;
}

const NavbarWrapper: NavbarStatic & FC<NavbarProps> = (props) => (
  <OpenProvider>
    <Navbar {...props} />
  </OpenProvider>
);

NavbarWrapper.Action = NavbarAction;

export default NavbarWrapper;
export { navbarLinkStyles } from './NavbarLink/NavbarLink';
