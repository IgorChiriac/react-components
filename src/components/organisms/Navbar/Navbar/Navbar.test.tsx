import React from 'react';
import { render, screen } from '@testing-library/react';
import Navbar from '..';

describe('<Navbar />', () => {
  const NAV_ITEMS = [
    {
      label: 'About',
      qadata: 'About.topBar.Menu',
      children: [
        {
          label: 'About Zopa',
          href: '/about',
          qadata: 'About.About.topBar.Menu',
        },
        {
          label: 'Our story',
          href: '/about/our-story',
          qadata: 'Story.About.topBar.Menu',
        },
        {
          label: 'Our board',
          href: '/about/board',
          qadata: 'Our_Board.About.topBar.Menu',
        },
        {
          label: 'Our leadership team',
          href: '/about/leadership',
          qadata: 'Our_Leadership.About.topBar.Menu',
        },
        {
          label: 'Awards',
          href: '/about/awards',
          qadata: 'Awards.About.topBar.Menu',
        },
        {
          label: 'Careers',
          href: '/about/careers',
          qadata: 'Careers.About.topBar.Menu',
        },
        {
          label: 'Press office',
          href: '/about/press',
          qadata: 'Press_office.About.topBar.Menu',
        },
      ],
    },
    {
      label: 'Support',
      href: '/contact',
      qadata: 'Support.topBar.Menu',
    },
  ];

  it('should render small device navigation with default props', () => {
    const { container } = render(<Navbar links={NAV_ITEMS} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render the hamburger icon when no content is added', () => {
    render(<Navbar />);
    const hamburgerIcon = screen.queryByTestId('hamburger-icon');
    expect(hamburgerIcon).not.toBeInTheDocument();
  });

  it('should render component with all the props', () => {
    const { container } = render(<Navbar overlayLogoWith={<a href="https://www.zopa.com"></a>} links={NAV_ITEMS} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render large device navigation with default props', () => {
    Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 1280 });

    const { container } = render(<Navbar links={NAV_ITEMS} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
