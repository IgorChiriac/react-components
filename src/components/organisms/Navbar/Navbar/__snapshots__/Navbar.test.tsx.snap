// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render component with all the props 1`] = `
.c4 {
  -webkit-transition: fill 0.3s ease;
  transition: fill 0.3s ease;
  fill: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  width: 100%;
}

.c3 {
  position: relative;
}

.c3 > a,
.c3 > button,
.c3 > div {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  width: 64px;
  font-size: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  width: 64px;
  font-size: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
}

.c7 {
  position: fixed;
  right: 0;
  top: 64px;
  bottom: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  width: 100%;
  padding: 40px 16px 0;
  background: #FFFFFF;
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  z-index: 2;
  overflow-y: auto;
}

.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px 24px;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-size: 15px;
  line-height: 1.2;
  font-weight: 600;
  cursor: pointer;
  border-radius: 8px;
  color: #FFFFFF;
  border: 1px solid transparent;
  outline: 0;
  background-color: #3B46C4;
  background: linear-gradient(90deg,#4F5AD8 0%,#3B46C4 100%);
}

.c6:hover:not(:disabled) {
  background: linear-gradient(90deg,#3B46C4 0%,#2732B0 100%);
}

.c6:focus:not(:active) {
  border: 1px solid #FFFFFF;
  box-shadow: 0 0 4px #3B46C4;
}

.c6:disabled {
  cursor: not-allowed;
  background: #F7F7F7;
  color: #818F9B;
}

.c6:active:not(:disabled) {
  border: 1px solid transparent;
  box-shadow: unset;
  opacity: 0.8;
}

@media (min-width:1280px) {
  .c0 {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;
    background-color: #FFFFFF;
    max-height: 120px;
    -webkit-transition: max-height 0.3s ease;
    transition: max-height 0.3s ease;
  }
}

@media (max-width:1280px) {
  .c0 .headroom {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;
    background-color: #00D9C5;
    -webkit-transition: -webkit-transform 200ms ease-in-out;
    -webkit-transition: transform 200ms ease-in-out;
    transition: transform 200ms ease-in-out;
  }

  .c0 .headroom--unfixed {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  .c0 .headroom--unpinned {
    -webkit-transform: translateY(-100%);
    -ms-transform: translateY(-100%);
    transform: translateY(-100%);
  }

  .c0 .headroom--pinned {
    -webkit-transform: translateY(0%);
    -ms-transform: translateY(0%);
    transform: translateY(0%);
  }
}

@media (min-width:1280px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: 490px;
    -webkit-transition: 0.3s min-height ease;
    transition: 0.3s min-height ease;
    min-height: 120px;
    padding-left: 64px;
  }
}

@media (min-width:1280px) {
  .c3:before {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    -webkit-transition: opacity 0.3s ease;
    transition: opacity 0.3s ease;
    background-image: url(nav-curve.svg);
    background-repeat: no-repeat;
    opacity: 1;
    content: '';
    z-index: -1;
  }
}

<header
  class="c0"
  role="banner"
>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
    >
      <nav
        class="c1"
      >
        <span
          class="c2"
          data-testid="hamburger-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-w-14 fa-fw "
            color="#FFFFFF"
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="c3"
        >
          <svg
            aria-label="zopa logo"
            color="#00D9C5"
            height="100%"
            role="img"
            version="1.1"
            viewBox="0 0 104 19"
            width="150px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              Zopa Logo
            </title>
            <g>
              <path
                class="c4"
                d="M36.90752,0 C42.6769778,0 46.8674844,3.93460861 46.8674844,9.36414355 C46.8674844,14.8199217 42.6769778,18.83326 36.90752,18.83326 C31.1380622,18.83326 26.9475556,14.8461649 26.9475556,9.36414355 C26.9475556,3.90836539 31.1380622,0 36.90752,0 Z M36.9851733,13.7701929 C39.0550044,13.7701929 40.7106844,11.9078615 40.7106844,9.38944951 C40.7106844,6.87197483 39.0550044,5.06212985 36.9851733,5.06212985 C34.8894578,5.06212985 33.1043556,6.87197483 33.1043556,9.38944951 C33.1043556,11.9078615 34.9153422,13.7701929 36.9851733,13.7701929 Z M16.4541867,1.04879158 C16.4541867,6.67702519 8.45959111,11.8600613 8.45959111,13.9529582 L16.1435733,13.9529582 L16.1435733,18.5708278 L0,18.5708278 L0,17.7048015 C0,12.0765679 7.99367111,6.89353176 7.99367111,4.8006349 L0.309688889,4.8006349 L0.309688889,0.183702546 L16.4541867,0.183702546 L16.4541867,1.04879158 Z M66.7153067,0.183702546 C71.4244267,0.183702546 74.1404444,2.56996113 74.1404444,6.76700194 C74.1404444,11.2255377 71.4244267,13.7964361 66.7153067,13.7964361 L64.2063644,13.7964361 L64.2063644,18.5708278 L58.5136356,18.5708278 L58.5136356,0.183702546 L66.7153067,0.183702546 Z M66.4046933,9.180441 C67.6989156,9.180441 68.44864,8.36690115 68.44864,6.95070449 C68.44864,5.58699426 67.6989156,4.79969764 66.4046933,4.79969764 L64.2063644,4.79969764 L64.2063644,9.180441 L66.4046933,9.180441 Z M96.6257067,0.612874778 L104,19 L97.6093156,19 L96.70336,16.4028584 L89.8994489,16.4028584 L88.9685333,19 L82.8108089,19 L90.4689067,0.612874778 L96.6257067,0.612874778 Z M91.4515911,11.619186 L95.2288711,11.619186 L93.3661156,6.24213753 L91.4515911,11.619186 Z"
              />
            </g>
          </svg>
          <a
            href="https://www.zopa.com"
          />
        </div>
        <span
          class="c5"
        >
          <button
            class="c6 ml-2"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-user fa-w-14 fa-fw "
              color="#FFFFFF"
              data-icon="user"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                fill="currentColor"
              />
            </svg>
          </button>
        </span>
        <aside
          class="c7"
        >
          Content
        </aside>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<Navbar /> should render large device navigation with default props 1`] = `
.c3 {
  -webkit-transition: fill 0.3s ease;
  transition: fill 0.3s ease;
  fill: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  width: 100%;
}

.c2 {
  position: relative;
}

.c2 > a,
.c2 > button,
.c2 > div {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px 24px;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-size: 15px;
  line-height: 1.2;
  font-weight: 600;
  cursor: pointer;
  border-radius: 8px;
  color: #FFFFFF;
  border: 1px solid transparent;
  outline: 0;
  background-color: #3B46C4;
  background: linear-gradient(90deg,#4F5AD8 0%,#3B46C4 100%);
}

.c5:hover:not(:disabled) {
  background: linear-gradient(90deg,#3B46C4 0%,#2732B0 100%);
}

.c5:focus:not(:active) {
  border: 1px solid #FFFFFF;
  box-shadow: 0 0 4px #3B46C4;
}

.c5:disabled {
  cursor: not-allowed;
  background: #F7F7F7;
  color: #818F9B;
}

.c5:active:not(:disabled) {
  border: 1px solid transparent;
  box-shadow: unset;
  opacity: 0.8;
}

@media (min-width:1280px) {
  .c0 {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;
    background-color: #FFFFFF;
    max-height: 120px;
    -webkit-transition: max-height 0.3s ease;
    transition: max-height 0.3s ease;
  }
}

@media (max-width:1280px) {
  .c0 .headroom {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;
    background-color: #00D9C5;
    -webkit-transition: -webkit-transform 200ms ease-in-out;
    -webkit-transition: transform 200ms ease-in-out;
    transition: transform 200ms ease-in-out;
  }

  .c0 .headroom--unfixed {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  .c0 .headroom--unpinned {
    -webkit-transform: translateY(-100%);
    -ms-transform: translateY(-100%);
    transform: translateY(-100%);
  }

  .c0 .headroom--pinned {
    -webkit-transform: translateY(0%);
    -ms-transform: translateY(0%);
    transform: translateY(0%);
  }
}

@media (min-width:1280px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: 490px;
    -webkit-transition: 0.3s min-height ease;
    transition: 0.3s min-height ease;
    min-height: 120px;
    padding-left: 64px;
  }
}

@media (min-width:1280px) {
  .c2:before {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    -webkit-transition: opacity 0.3s ease;
    transition: opacity 0.3s ease;
    background-image: url(nav-curve.svg);
    background-repeat: no-repeat;
    opacity: 1;
    content: '';
    z-index: -1;
  }
}

@media (min-width:1280px) {
  .c4 {
    margin-right: 64px;
  }
}

<header
  class="c0"
  role="banner"
>
  <nav
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        aria-label="zopa logo"
        height="100%"
        role="img"
        version="1.1"
        viewBox="0 0 104 19"
        width="150px"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <title>
          Zopa Logo
        </title>
        <g>
          <path
            class="c3"
            d="M36.90752,0 C42.6769778,0 46.8674844,3.93460861 46.8674844,9.36414355 C46.8674844,14.8199217 42.6769778,18.83326 36.90752,18.83326 C31.1380622,18.83326 26.9475556,14.8461649 26.9475556,9.36414355 C26.9475556,3.90836539 31.1380622,0 36.90752,0 Z M36.9851733,13.7701929 C39.0550044,13.7701929 40.7106844,11.9078615 40.7106844,9.38944951 C40.7106844,6.87197483 39.0550044,5.06212985 36.9851733,5.06212985 C34.8894578,5.06212985 33.1043556,6.87197483 33.1043556,9.38944951 C33.1043556,11.9078615 34.9153422,13.7701929 36.9851733,13.7701929 Z M16.4541867,1.04879158 C16.4541867,6.67702519 8.45959111,11.8600613 8.45959111,13.9529582 L16.1435733,13.9529582 L16.1435733,18.5708278 L0,18.5708278 L0,17.7048015 C0,12.0765679 7.99367111,6.89353176 7.99367111,4.8006349 L0.309688889,4.8006349 L0.309688889,0.183702546 L16.4541867,0.183702546 L16.4541867,1.04879158 Z M66.7153067,0.183702546 C71.4244267,0.183702546 74.1404444,2.56996113 74.1404444,6.76700194 C74.1404444,11.2255377 71.4244267,13.7964361 66.7153067,13.7964361 L64.2063644,13.7964361 L64.2063644,18.5708278 L58.5136356,18.5708278 L58.5136356,0.183702546 L66.7153067,0.183702546 Z M66.4046933,9.180441 C67.6989156,9.180441 68.44864,8.36690115 68.44864,6.95070449 C68.44864,5.58699426 67.6989156,4.79969764 66.4046933,4.79969764 L64.2063644,4.79969764 L64.2063644,9.180441 L66.4046933,9.180441 Z M96.6257067,0.612874778 L104,19 L97.6093156,19 L96.70336,16.4028584 L89.8994489,16.4028584 L88.9685333,19 L82.8108089,19 L90.4689067,0.612874778 L96.6257067,0.612874778 Z M91.4515911,11.619186 L95.2288711,11.619186 L93.3661156,6.24213753 L91.4515911,11.619186 Z"
          />
        </g>
      </svg>
    </div>
    <div
      class="c4"
    >
      Content
      <button
        class="c5 ml-2"
      >
        Sign-in
      </button>
    </div>
  </nav>
</header>
`;

exports[`<Navbar /> should render small device navigation with default props 1`] = `
.c4 {
  -webkit-transition: fill 0.3s ease;
  transition: fill 0.3s ease;
  fill: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  width: 100%;
}

.c3 {
  position: relative;
}

.c3 > a,
.c3 > button,
.c3 > div {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  width: 64px;
  font-size: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  width: 64px;
  font-size: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
}

.c7 {
  position: fixed;
  right: 0;
  top: 64px;
  bottom: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  width: 100%;
  padding: 40px 16px 0;
  background: #FFFFFF;
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  z-index: 2;
  overflow-y: auto;
}

.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px 24px;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-size: 15px;
  line-height: 1.2;
  font-weight: 600;
  cursor: pointer;
  border-radius: 8px;
  color: #FFFFFF;
  border: 1px solid transparent;
  outline: 0;
  background-color: #3B46C4;
  background: linear-gradient(90deg,#4F5AD8 0%,#3B46C4 100%);
}

.c6:hover:not(:disabled) {
  background: linear-gradient(90deg,#3B46C4 0%,#2732B0 100%);
}

.c6:focus:not(:active) {
  border: 1px solid #FFFFFF;
  box-shadow: 0 0 4px #3B46C4;
}

.c6:disabled {
  cursor: not-allowed;
  background: #F7F7F7;
  color: #818F9B;
}

.c6:active:not(:disabled) {
  border: 1px solid transparent;
  box-shadow: unset;
  opacity: 0.8;
}

@media (min-width:1280px) {
  .c0 {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;
    background-color: #FFFFFF;
    max-height: 120px;
    -webkit-transition: max-height 0.3s ease;
    transition: max-height 0.3s ease;
  }
}

@media (max-width:1280px) {
  .c0 .headroom {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;
    background-color: #00D9C5;
    -webkit-transition: -webkit-transform 200ms ease-in-out;
    -webkit-transition: transform 200ms ease-in-out;
    transition: transform 200ms ease-in-out;
  }

  .c0 .headroom--unfixed {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  .c0 .headroom--unpinned {
    -webkit-transform: translateY(-100%);
    -ms-transform: translateY(-100%);
    transform: translateY(-100%);
  }

  .c0 .headroom--pinned {
    -webkit-transform: translateY(0%);
    -ms-transform: translateY(0%);
    transform: translateY(0%);
  }
}

@media (min-width:1280px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: 490px;
    -webkit-transition: 0.3s min-height ease;
    transition: 0.3s min-height ease;
    min-height: 120px;
    padding-left: 64px;
  }
}

@media (min-width:1280px) {
  .c3:before {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    -webkit-transition: opacity 0.3s ease;
    transition: opacity 0.3s ease;
    background-image: url(nav-curve.svg);
    background-repeat: no-repeat;
    opacity: 1;
    content: '';
    z-index: -1;
  }
}

<header
  class="c0"
  role="banner"
>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
    >
      <nav
        class="c1"
      >
        <span
          class="c2"
          data-testid="hamburger-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-w-14 fa-fw "
            color="#FFFFFF"
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="c3"
        >
          <svg
            aria-label="zopa logo"
            color="#00D9C5"
            height="100%"
            role="img"
            version="1.1"
            viewBox="0 0 104 19"
            width="150px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              Zopa Logo
            </title>
            <g>
              <path
                class="c4"
                d="M36.90752,0 C42.6769778,0 46.8674844,3.93460861 46.8674844,9.36414355 C46.8674844,14.8199217 42.6769778,18.83326 36.90752,18.83326 C31.1380622,18.83326 26.9475556,14.8461649 26.9475556,9.36414355 C26.9475556,3.90836539 31.1380622,0 36.90752,0 Z M36.9851733,13.7701929 C39.0550044,13.7701929 40.7106844,11.9078615 40.7106844,9.38944951 C40.7106844,6.87197483 39.0550044,5.06212985 36.9851733,5.06212985 C34.8894578,5.06212985 33.1043556,6.87197483 33.1043556,9.38944951 C33.1043556,11.9078615 34.9153422,13.7701929 36.9851733,13.7701929 Z M16.4541867,1.04879158 C16.4541867,6.67702519 8.45959111,11.8600613 8.45959111,13.9529582 L16.1435733,13.9529582 L16.1435733,18.5708278 L0,18.5708278 L0,17.7048015 C0,12.0765679 7.99367111,6.89353176 7.99367111,4.8006349 L0.309688889,4.8006349 L0.309688889,0.183702546 L16.4541867,0.183702546 L16.4541867,1.04879158 Z M66.7153067,0.183702546 C71.4244267,0.183702546 74.1404444,2.56996113 74.1404444,6.76700194 C74.1404444,11.2255377 71.4244267,13.7964361 66.7153067,13.7964361 L64.2063644,13.7964361 L64.2063644,18.5708278 L58.5136356,18.5708278 L58.5136356,0.183702546 L66.7153067,0.183702546 Z M66.4046933,9.180441 C67.6989156,9.180441 68.44864,8.36690115 68.44864,6.95070449 C68.44864,5.58699426 67.6989156,4.79969764 66.4046933,4.79969764 L64.2063644,4.79969764 L64.2063644,9.180441 L66.4046933,9.180441 Z M96.6257067,0.612874778 L104,19 L97.6093156,19 L96.70336,16.4028584 L89.8994489,16.4028584 L88.9685333,19 L82.8108089,19 L90.4689067,0.612874778 L96.6257067,0.612874778 Z M91.4515911,11.619186 L95.2288711,11.619186 L93.3661156,6.24213753 L91.4515911,11.619186 Z"
              />
            </g>
          </svg>
        </div>
        <span
          class="c5"
        >
          <button
            class="c6 ml-2"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-user fa-w-14 fa-fw "
              color="#FFFFFF"
              data-icon="user"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                fill="currentColor"
              />
            </svg>
          </button>
        </span>
        <aside
          class="c7"
        >
          Content
        </aside>
      </nav>
    </div>
  </div>
</header>
`;
